app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.kt

```diff
@@ -87,61 +88,57 @@ object LicenseFragmentHelper {
-                    alert.setView(webView)
-                    Localization.assureCorrectAppLanguage(context)
-                    alert.setNegativeButton(
-                        context.getString(R.string.ok)
-                    ) { dialog, _ -> dialog.dismiss() }
-                    alert.show()
+                .subscribe { formattedLicense ->
+                    AlertDialog.Builder(context).apply {
+                        setTitle(license.name)
+                        setView(loadLicense(context, formattedLicense))
+                        Localization.assureCorrectAppLanguage(context)
+                        setPositiveButton(R.string.ok) { dialog, _ ->
+                            dialog.dismiss()
+                        }
+                        show()
+                    }
                 }
         }
     }
-    @JvmStatic
+
     fun showLicense(context: Context?, component: SoftwareComponent): Disposable {
         return if (context == null) {
             Disposable.empty()
         } else {
             Observable.fromCallable { getFormattedLicense(context, component.license) }
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
-                .subscribe { formattedLicense: String ->
-                    val webViewData = Base64.encodeToString(
-                        formattedLicense
-                            .toByteArray(StandardCharsets.UTF_8),
-                        Base64.NO_PADDING
-                    )
-                    val webView = WebView(context)
-                    webView.loadData(webViewData, "text/html; charset=UTF-8", "base64")
-                    val alert = AlertDialog.Builder(context)
-                    alert.setTitle(component.license.name)
-                    alert.setView(webView)
-                    Localization.assureCorrectAppLanguage(context)
-                    alert.setPositiveButton(
-                        R.string.dismiss
-                    ) { dialog, _ -> dialog.dismiss() }
-                    alert.setNeutralButton(R.string.open_website_license) { _, _ ->
-                        ShareUtils.openUrlInBrowser(context, component.link)
+                .subscribe { formattedLicense ->
+                    AlertDialog.Builder(context).apply {
+                        setTitle(component.license.name)
+                        setView(loadLicense(context, formattedLicense))
+                        Localization.assureCorrectAppLanguage(context)
+                        setPositiveButton(R.string.dismiss) { dialog, _ ->
+                            dialog.dismiss()
+                        }
+                        setNeutralButton(R.string.open_website_license) { _, _ ->
+                            ShareUtils.openUrlInBrowser(context, component.link)
+                        }
+                        show()
                     }
-                    alert.show()
                 }
         }
     }
```

Comment:
These methods are nearly identical.
Can they be merged into one method?

(also then we wouldn't need ``loadLicense`` below)

Reply:
Uh, perhaps? I'll see what I can do.

Reply:
Update: Kotlin is a godsend.